python/utils/bootstrap_utilities.py

```diff
@@ -31,9 +31,23 @@ def start_engine(data):
     # a current user, we have to set the authenticated user.
     if hasattr(sgtk, "set_authenticated_user"):
         # Retrieve the currently authenticated user for this process.
-        from tank_vendor.shotgun_authentication import ShotgunAuthenticator
-        user = ShotgunAuthenticator(sgtk.util.CoreDefaultsManager()).get_default_user()
-        sgtk.set_authenticated_user(user)
+        from tank_vendor.shotgun_authentication import ShotgunAuthenticator, deserialize_user
+        current_user = ShotgunAuthenticator(sgtk.util.CoreDefaultsManager()).get_default_user()
+
+        # If we found no user using the authenticator, we need to use the credentials that
```

Comment:
Why is `SHOTGUN_DESKTOP_CURRENT_USER` a fallback? is it a performance thing? Shouldn't we always look for this and pick it up and use it? like a contract between the two RPC desktop processes? It might be worth adding a comment to clarify why we need to pick up the default user!